table/sst_file_writer.cc

```diff
@@ -90,6 +94,11 @@ struct SstFileWriter::Rep {
 
     builder->Add(ikey.Encode(), value);
 
+    Status s = kv_validator->Add(ikey.Encode(), value);
+    if(!s.ok()){
+      return s;
+    }
```

Comment:
Can we eliminate the check on lines 79-84 now that we have this?

Reply:
Hey, sorry for my late response too. For my own understanding, what's the purpose of reading back the data and comparing the hashes, is it probably for extra fidelity?

Reply:
Thinking about performance, wouldn't it be computationally expensive to read the entire SST for every new write ? Or would users be ready to pay the costs when they set `paranoid_file_checks` to true ?

Reply:
@ajkr Thanks for the tip. About reading the data back, I haven't found an API to read the entries entered into a table builder. From the code snippet you shared, I think it's quite an overhead to have to go through the cache just to read the data back (I'm happy to be wrong about this). So my question is - is there an API that I can conveniently use to read the data entered into a table reader, or does this warrant us to build that API, which I'm happy to do.